File: library/src/main/java/com/getstream/sdk/chat/style/ChatStyle.java
package com.getstream.sdk.chat.style;

import androidx.annotation.FontRes;

public interface ChatStyle {

    TextStyle getDefaultTextStyle();

    boolean hasDefaultFont();

    class Builder {

        ChatStyleImpl result = new ChatStyleImpl();
        TextStyle defaultTextStyle;

        public Builder setDefaultFont(String assetPath) {
            defaultTextStyle = new TextStyle();
            defaultTextStyle.fontAssetsPath = assetPath;
            return this;
        }

        public Builder setDefaultFont(@FontRes int fontRes) {
            defaultTextStyle = new TextStyle();
            defaultTextStyle.fontResource = fontRes;
            return this;
        }

        public ChatStyle build() {
            result.defaultTextStyle = defaultTextStyle;
            return result;
        }
    }
}

File: library/src/main/java/com/getstream/sdk/chat/style/ChatStyle.kt
package com.getstream.sdk.chat.style

import androidx.annotation.FontRes

interface ChatStyle {

    fun hasDefaultFont(): Boolean

    fun getDefaultTextStyle(): TextStyle

    class Builder {
        var result = ChatStyleImpl()

        fun setDefaultFont(assetPath: String): Builder = apply {
            result.defaultTextStyle = TextStyle().apply {
                fontAssetsPath = assetPath
            }
        }

        fun setDefaultFont(@FontRes fontRes: Int): Builder = apply {
            result.defaultTextStyle = TextStyle().apply {
                fontResource = fontRes
            }
        }

        fun build(): ChatStyle = result
    }
}
